doc/lua_api.txt

```diff
@@ -3303,21 +3299,67 @@ The following functions provide escape sequences:
 
 Spatial Vectors
 ===============
-A spatial vector is similar to a position, but instead using
-absolute world coordinates, it uses *relative* coordinates, relative to
-no particular point.
-
-Internally, it is implemented as a table with the 3 fields
-`x`, `y` and `z`. Example: `{x = 0, y = 1, z = 0}`.
-However, one should *never* create a vector manually as above, such misbehavior
-is deprecated. The vector helpers set a metatable for the created vectors which
-allows indexing with numbers, calling functions directly on vectors and using
-operators (like `+`). Furthermore, the internal implementation might change in
+
+Minetest stores 3-dimensional spatial vectors in Lua as tuples of 3 coordinates,
```

Comment:
They aren't really tuples (= varargs or lists you could say), but rather as tables (hash map). I don't like this much abstraction.

Reply:
Changed to "tables".
But I don't really like defining vectors as tables here. It kinda leaks an implementation detail. But I guess it's ok.
One could also say "struct" or "[record](https://en.wikipedia.org/wiki/Record_(computer_science))", as using a hash-table-like table is the idiomatic way to do structs in lua (afaik), but then that's again the same as "tuple".
(Btw. IIRC last time I tried, very often indexing a small array table was about as fast as with a hash table, in LuaJIT with jit off.)